xterm/shellserver.pl

```diff
@@ -68,13 +68,17 @@
 &reset_environment();
 my $shcmd = "'$shellexec".($shelllogin ? " $shelllogin" : "")."'";
 if (!$pid) {
-	&cleanup_miniserv();
+	&remove_miniserv_websocket($port);
 	die "Failed to run shell with $shcmd\n";
 	}
 else {
 	print STDERR "Running shell $shcmd with pid $pid\n";
 	}
 
+# Save new websocket info to miniserv.conf
+# only after we get the pty pid
+&save_miniserv_websocket($port);
```

Comment:
We can't move this down, because the port allocation and updating miniserv.conf has to be done while the file is locked, to prevent a collision between multiple shells launched at the same time. That's why the current code has it all in one block.

Reply:
Alright, fixed.